sql/catalyst/src/main/java/org/apache/spark/sql/execution/UnsafeExternalRowSorter.java

```diff
@@ -211,7 +211,10 @@ public int compare(Object baseObj1, long baseOff1, Object baseObj2, long baseOff
       // TODO: Why are the sizes -1?
       row1.pointTo(baseObj1, baseOff1, -1);
       row2.pointTo(baseObj2, baseOff2, -1);
-      return ordering.compare(row1, row2);
+      int comparison = ordering.compare(row1, row2);
+      row1.pointTo(null, 0L, -1);
+      row2.pointTo(null, 0L, -1);
```

Comment:
Good idea @cloud-fan . Looks like `RowComparator` and `KVComparator` could be cleaned up in `UnsafeExternalSorter.cleanupResources` and `UnsafeInMemorySorter.free`. @davies I know this is from a long while back, but does that make sense? Seems like reasonable places to simply 'flush' the references, and won't hurt anything.

I am not 100% sure there's not another path to take care of, or if this frees the ref soon enough to avoid the problem. Thoughts @j-baker ?